ingest_openapi.yaml

```diff
@@ -100,12 +123,65 @@ components:
               - genomic_id
               - access_method
               - samples
+              - index
+    ClinicalDonorRequest:
+      content:
+        'application/json':
+          schema:
+            $ref: "#/components/schemas/DonorWithClinicalDataField"
   schemas:
     S3Access:
       type: string
       description: a specification for an S3 bucket
       pattern: s3:\/\/(.+)\/(.+)
       example: s3://http://candig.docker.internal:9000/mohccn-data
+    DonorWithClinicalDataField: # "Sanity check" validation, everyting else happens in ETL validation/katsu
```

Comment:
This schema seems wrong: it suggests that a DonorWithClinicalDataField is an array of named properties? Also that the named properties are all themselves DonorWithClinicalDataFields? Isn't this circular? What is the point of this schema?

I don't know why my OpenAPI editor doesn't flag this as problematic, but my understanding is that arrays shouldn't have properties, because properties are names of keys of objects. Arrays should have unnamed items, which are of particular types.

See my comment on line 131.

Reply:
Sorry this is a little confusing, effectively DonorWithClinicalData is an array with a list of objects, each of which must follow the specification under the properties tag. OpenAPI just implicitly provides the type: object, but I've added that explicitly in [0dda517](https://github.com/CanDIG/candigv2-ingest/pull/27/commits/0dda51760049870ff1dc2fc47e37473cbe25f463). The idea with those circular references is that only sub-fields of objects (e.g. treatments) can have a dictionary representing that subfield as their value (which can contain further subfields), any other key containing a dictionary will raise red flags. Again it's just supposed to be a very basic sanity check, the code/Katsu should detect the rest.